File: workout-api/src/main/java/tom/projects/monkeylog/mapper/MeasurementMapper.java
package tom.projects.monkeylog.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;
import org.mapstruct.ReportingPolicy;
import tom.projects.monkeylog.dto.measurement.*;
import tom.projects.monkeylog.model.measurement.Measurement;
import tom.projects.monkeylog.model.measurement.MeasurementPoint;

import java.util.List;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface MeasurementMapper {
    MeasurementFullResponse measurementToMeasurementFullResponse(Measurement measurement);

    List<MeasurementFullResponse> measurementsToMeasurementFullResponses(List<Measurement> measurementList);

    MeasurementResponse measurementToMeasurementResponse(Measurement measurement);

    MeasurementPointResponse measurementPointToMeasurementPointResponse(MeasurementPoint measurementPoint);

    Measurement measurementCreateRequestToMeasurement(MeasurementCreateRequest measurementcreateRequest);

    void updateMeasurement(@MappingTarget Measurement measurement, MeasurementUpdateRequest measurementUpdateRequest);
}

File: backendkt/src/main/kotlin/tom/com/monkeylog/mapper/MeasurementMapper.kt
package tom.com.monkeylog.mapper

import org.mapstruct.Mapper
import org.mapstruct.MappingTarget
import org.mapstruct.ReportingPolicy
import tom.com.monkeylog.dto.measurement.*
import tom.com.monkeylog.model.measurement.Measurement
import tom.com.monkeylog.model.measurement.MeasurementPoint

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
interface MeasurementMapper {
    fun measurementToMeasurementFullResponse(measurement: Measurement): MeasurementFullResponse
    fun measurementsToMeasurementFullResponses(measurementList: List<Measurement>): List<MeasurementFullResponse>
    fun measurementToMeasurementResponse(measurement: Measurement): MeasurementResponse
    fun measurementPointToMeasurementPointResponse(measurementPoint: MeasurementPoint): MeasurementPointResponse
    fun measurementCreateRequestToMeasurement(measurementCreateRequest: MeasurementCreateRequest): Measurement
    fun updateMeasurement(@MappingTarget measurement: Measurement, measurementUpdateRequest: MeasurementUpdateRequest)
}
